my_game.py

```diff
@@ -214,8 +214,10 @@ def on_update(self, delta_time):
         # Move player on x or y axis
         if self.left_pressed and not self.right_pressed:
             self.player.change_x = -PLAYER_SPEED
+            self.player.width = -17
         elif self.right_pressed and not self.left_pressed:
             self.player.change_x = PLAYER_SPEED
+            self.player.width = 17
```

Comment:
This will not work if the sprite uses a graphic of a different size. You should flip the graphic instead.

Reply:
That doesn't work. I tried it. But it only works when I created the object of the player. I can't seem to make it flip like that. Trust me. I tried that. A lot. Could we work on another solution?

Reply:
> That doesn't work. I tried it. But it only works when I created the object of the player. I can't seem to make it flip like that. Trust me. I tried that. A lot. Could we work on another solution?

Show me the code

Reply:
I

> > That doesn't work. I tried it. But it only works when I created the object of the player. I can't seem to make it flip like that. Trust me. I tried that. A lot. Could we work on another solution?
> 
> Show me the code

It's not that. It's just that it doesn't do anything. We're insensibly making a new variable for the self.player. It doesn't know what "flipped horizontally" is. It's not a thing you can call and make it a bool.

Reply:
You are right... I see it now. You have to load the graphics twice, and flip one. Then switch between them. This should happen in the _Player_ class. You need to implement the switching between textures by adding the method _update()_ like you can see in the following example:

https://api.arcade.academy/en/latest/examples/sprite_face_left_or_right.html#sprite-face-left-or-right

Reply:
Actually, I think we need to move the knowledge of what keys move the player(s) into the player class. Then we can flip the graphic when we change direction from within the player object...

Reply:
Deal, I'll wait until you merged yours into it. And then I'll do that. 👍